test.cc

```diff
@@ -136,27 +158,70 @@ TEST(Sherlock, SubscribeAndProcessThreeEntriesByUniquePtr) {
   bar_stream.Publish(5);
   bar_stream.Publish(6);
   Data d;
-  std::unique_ptr<Processor> p(new Processor(d));
+  ASSERT_FALSE(d.listener_alive_);
+  std::unique_ptr<Processor> p(new Processor(d, false));
+  ASSERT_TRUE(d.listener_alive_);
   p->SetMax(3u);
-  bar_stream.Subscribe(std::move(p))
-      .Join();  // `.Join()` makes this a blocking call, waiting for three entries.
-  EXPECT_EQ(d.seen_, 3u);
-  EXPECT_EQ("4,5,6", d.results_);
+  bar_stream.AsyncSubscribe(std::move(p)).Join();  // `.Join()` blocks this thread waiting for three entries.
+  EXPECT_EQ(3u, d.seen_);
+  while (d.listener_alive_) {
+    ;  // Spin lock.
+  }
+
+  // A careful condition, since the listener may process some or all entries before going out of scope.
+  EXPECT_TRUE((d.results_ == "TERMINATE,4,5,6") || (d.results_ == "4,TERMINATE,5,6") ||
+              (d.results_ == "4,5,TERMINATE,6") || (d.results_ == "4,5,6,TERMINATE") || (d.results_ == "4,5,6"))
+      << d.results_;
```

Comment:
It will dump the contents of that `d.results_` string to `stdout` (or `stderr`) if the condition fails.

Google Test library also acts as Google Logging library, allowing each `ASSERT` or `EXPECT` to act as an output stream for debug info. [PFM](http://www.allacronyms.com/PFM/Pure_Fucking_Magic).